oaknational/Oak-Web-Application
# Feat/pupil 570/pupil 628/pupil subject page

## Description

Music year: {owa_music_year}

adds Subject query to OWA

create subjct listing page for year group defaults to !isLegacy

each subject is a link to either the unit-lisitng or programme page if available


- [ x]  A list of available subjects is provided with slug, and display title
- [x ]  subjects with programme factors have a field to indicate this
- [x ]  The list is filterable by is_legacy
- [ x]  list of links shown with correct titles and URLs
- [ x]  URL pointing to programme page where there are programme factors
- [ ]  tests for programme logic

## Issue(s)

Fixes #

## How to test

1. Go to {owa_deployment_url}/pupils/beta/years/year-10/subjects
2. Click on subject 
3. You should see if there are no other programme factors you are taken to unit listing/ programme listing if not

## Screenshots

How it used to look (delete if n/a):
{screenshots}

How it should now look:
{screenshots}

## Checklist
